azblob/url_blob.go

```diff
@@ -143,6 +143,16 @@ func (b BlobURL) Delete(ctx context.Context, deleteOptions DeleteSnapshotsOption
 		nil, BlobDeleteNone)
 }
 
+// PermanentDelete permanently deletes-soft deleted snapshots & soft-deleted version blobs and is a dangerous operation and SHOULD NOT BE USED.
+// For more information, see https://docs.microsoft.com/rest/api/storageservices/delete-blob.
+func (b BlobURL) PermanentDelete(ctx context.Context, deleteOptions DeleteSnapshotsOptionType, ac BlobAccessConditions, blobDeleteType BlobDeleteType) (*BlobDeleteResponse, error) {
+	ifModifiedSince, ifUnmodifiedSince, ifMatchETag, ifNoneMatchETag := ac.ModifiedAccessConditions.pointers()
+	return b.blobClient.Delete(ctx, nil, nil, nil, ac.LeaseAccessConditions.pointers(), deleteOptions,
+		ifModifiedSince, ifUnmodifiedSince, ifMatchETag, ifNoneMatchETag,
+		nil, // Blob ifTags
+		nil, blobDeleteType)
```

Comment:
shouldn't the delete type be set already? Since the function is already called permanent delete